sdks/java/core/src/main/java/org/apache/beam/sdk/io/DefaultFilenamePolicy.java

```diff
@@ -95,36 +149,67 @@ public static DefaultFilenamePolicy constructUsingStandardParameters(
    * strings.
    *
    * <p>Within a shard template, repeating sequences of the letters "S" or "N"
-   * are replaced with the shard number, or number of shards respectively.  The
-   * numbers are formatted with leading zeros to match the length of the
+   * are replaced with the shard number, or number of shards respectively.
+   * Repeating sequence of "P" is replaced with the index of the window pane.
+   * "L" is replaced with "true" in case of last pane or "false" otherwise.
+   * "F" is replaced with "true" in case of first pane or "false" otherwise.
```

Comment:
For instance, users who have daily calendar windows might want the window to just be the day - hour, minute, etc. are redundant to them. Some users might want to create separate directories for year, month, day, and only include time in the filename. There are a hundred variations, and we can't support them all here. It's perfectly fine to start off with a basic default, and expand later if needed.